opensrp-chw/src/main/java/org/smartregister/chw/activity/MalariaProfileActivity.java

```diff
@@ -335,6 +395,152 @@ public void onComplete() {
+
+            Rules rules = CoreChwApplication.getInstance().getRulesEngineHelper().rules(CoreConstants.RULE_FILE.PNC_HOME_VISIT);
+
+            PncVisitAlertRule pncVisitAlertRule = HomeVisitUtil.getPncVisitStatus(rules, getLastVisitDate(ancMemberObject.getBaseEntityId()), getDeliveryDate(ancMemberObject.getBaseEntityId()));
+
+            //check last visit
+            String lastVisit = getPncLastVisitDate(memberObject.getBaseEntityId());
+            if (lastVisit != null) {
+                int numOfDays = Days.daysBetween(new DateTime(formatDate(lastVisit)).toLocalDate(), new DateTime().toLocalDate()).getDays();
+                if (numOfDays <= 1) {
+                    visitDone.setVisibility(View.VISIBLE);
+                    textViewVisitDoneEdit.setTag("EDIT_PNC");
+                    textViewVisitDone.setTextColor(getResources().getColor(R.color.alert_complete_green));
+                    textViewRecordAnc.setVisibility(View.GONE);
+                } else {
+                    setPncViews(pncVisitAlertRule);
+                }
+            } else {
+                setPncViews(pncVisitAlertRule);
+            }
+        }
+
+    }
+
+    public Visit getVisit(String eventType) {
+        return getInstance().visitRepository().getLatestVisit(MEMBER_OBJECT.getBaseEntityId(), eventType);
+    }
+
+
+    protected String getPncLastVisitDate(String motherBaseID) {
+
+        Long pncLastVisitdate = PncLibrary.getInstance().profileRepository().getLastVisit(motherBaseID);
+        if (pncLastVisitdate != null) {
+            Date pncDate = new Date(pncLastVisitdate);
+            SimpleDateFormat format = new SimpleDateFormat("dd MMM", Locale.getDefault());
+            return format.format(pncDate);
+        }
+
+        return null;
+    }
+
+    private void saveVisit(String eventType) {
+        try {
+            Event event = org.smartregister.chw.anc.util.JsonFormUtils.createUntaggedEvent(MEMBER_OBJECT.getBaseEntityId(), eventType, org.smartregister.chw.anc.util.Constants.TABLES.ANC_MEMBERS);
+            Visit visit = NCUtils.eventToVisit(event, org.smartregister.chw.anc.util.JsonFormUtils.generateRandomUUIDString());
+            visit.setPreProcessedJson(new Gson().toJson(event));
+            getInstance().visitRepository().addVisit(visit);
+        } catch (JSONException e) {
+            Timber.e(e);
+        }
+    }
+
+    protected boolean ancHomeVisitNotDoneEvent(Visit visit) {
+
+        return visit != null
+                && (new DateTime(visit.getDate()).getMonthOfYear() == new DateTime().getMonthOfYear())
+                && (new DateTime(visit.getDate()).getYear() == new DateTime().getYear());
+    }
+
+    private Date getLastVisitDate(String baseId) {
+        Visit lastVisit = AncLibrary.getInstance().visitRepository().getLatestVisit(baseId, org.smartregister.chw.anc.util.Constants.EVENT_TYPE.PNC_HOME_VISIT);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'AncLibrary.getInstance' due to existing static import 'org.smartregister.chw.anc.AncLibrary.getInstance'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5426626)